src/seedu/addressbook/data/person/ContactDetail.java

```diff
@@ -0,0 +1,84 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a person's contact detail
+ * Guarantees: immutable; is valid as declared in {@link #isValid(String)}
+ */
+public abstract class ContactDetail {
```

Comment:
note that the super class should have been named `Contact` as per the exercise specification.